blockchain/consensus/deleg_cns/README.md

```diff
@@ -0,0 +1,225 @@
+# Delegated Consensus
+
+_Delegated Consensus_ is a simplistic consensus protocol made for demo purposes, where only a single miner is allowed to propose blocks - this miner is who the others "delegate to".
+
+## Setup Genesis
+
+To get started, we need to generate a `genesis.car` file that contains the public key of this miner, so that participants can validate that the blocks they receive are signed by the expected miner.
+
+At the time of this writing _Forest_ doesn't have the necessary machinery to prepare its own `genesis.car` file (although it has [commands](https://github.com/ChainSafe/forest/blob/3d149dcd7cfd23a8ce2793e107a4c7db20948584/forest/src/cli/genesis_cmd.rs#L19) to prepare a template for it), so for the time being we have to rely on [Lotus CLI](https://github.com/filecoin-project/lotus/tree/v1.17.0-rc3/cmd) to do this for us.
+
+
+### Build Lotus
+
+Ideally we could avoid having to build Lotus just for this purpose, and use the published [Docker image](https://hub.docker.com/r/filecoin/lotus). However, not all commands that we need are part of the image; for example `lotus-seed` is only available in the [lotus-test](https://github.com/filecoin-project/lotus/blob/v1.17.0-rc3/Dockerfile.lotus#L232) build target.
+
+Still, we can use docker to build the CLI without having to install all necessary build dependencies ourselves. Assuming that we already have Lotus cloned from Github and we are in the top level `lotus` directory, we can use the following command to build an image we need:
+
+```bash
+docker build -t filecoin/lotus-test -f Dockerfile.lotus --target lotus-test .
+```
+
+### Generate signing keys
+
+According to the [spec](https://spec.filecoin.io/#section-systems.filecoin_nodes.repository.key_store) we need to generate a BLS key for signing blocks, a.k.a. the "worker" key.
```

Comment:
The more I think about it, the more I feel we'll need to decouple the use of miner actors for HC validators in non Filecoin EC consensus. Even if we want to still use the concepts of worker and owner addresses, it'd be good to decouple into a new actor without all the storage-specific functions. 

For future iteration, it may be a good idea to just use account actors for validators in non-Filecoin EC consensus (I am aware this may require some changes to the validation logic, unfortunately).

Reply:
> We can't rock the boat too much if we want to keep in sync with the core Forest changes.

Definitely. I don't want to suffer again what we will potentially suffer when we push Eudico's changes to Lotus :) I feel that as the Forest team is smaller they will be happy with us contributing as long as we respect the compatibility with mainnet. I would definitely propose all the changes that we need (as long as they're non-breaking) and make Forest also "our thing". (Honestly, right now I think you are one of the top people with deep knowledge of the base code --which are few--)